src/Database/Connections/Connection.php

```diff
@@ -50,14 +48,31 @@ public function logQuery($query, $bindings, $time = null)
      * Fire an event for this connection.
      *
      * @param  string  $event
-     * @return void
+     * @return array|null
      */
     protected function fireConnectionEvent($event)
     {
-        if (isset($this->events)) {
-            $this->events->fire('connection.'.$this->getName().'.'.$event, $this);
-        }
+        $this->fireEvent('connection.'.$this->getName().'.'.$event, $this);
 
         parent::fireConnectionEvent($event);
     }
+
+    /**
+     * Fire the given event if possible.
+     *
+     * @param string $event
+     * @param array|object $attributes
+     * @return void
+     */
+    protected function fireEvent($event, $attributes = [])
```

Comment:
Is this a Laravel override? If not we can probably add the type hints direct to the signature instead of the docblock

Reply:
It's not, no. I'll add the type hints.